### PR TITLE
feat: Working application

### PR DESCRIPTION
**Pull request by JorgeAJT**

### CODE REVIEW

**Comment by dbrosatsmartworkz** [comment]
Este soy yo poniendo un comentario :)

File: db/meter_readings.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
 CREATE TABLE meter_readings
 (
-    meter_readings_id int PRIMARY KEY,
+    meter_readings_id SERIAL PRIMARY KEY,
```
Comment (dbrosatsmartworkz) [inline db/meter_readings.sql]: int GENERATED ALWAYS AS IDENTITY is better: https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-identity-column/

File: src/utils/insert_into_db.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        cursor = conn.cursor()
+
+        cursor.execute(f"""
+            SELECT column_name 
+            FROM information_schema.columns 
+            WHERE table_name = '{table_name}' 
+            AND column_default IS NULL
+            AND is_identity = 'NO'
+        """)
+        table_columns = [row[0] for row in cursor.fetchall()]
```
Comment (dbrosatsmartworkz) [inline src/utils/insert_into_db.py]: Why are you fetching columns before insert?

File: src/utils/insert_into_db.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                    value = json.dumps(value)
+                elif value is None:
+                    value = 'NULL'
+                values.append(value)
+            data_to_insert.append(values)
+
+        print(f"Insert statement: {insert_statement}")
+        print(f"Data to insert: {data_to_insert}")
+
+        execute_values(cursor, insert_statement, data_to_insert)
```
Comment (dbrosatsmartworkz) [inline src/utils/insert_into_db.py]: Why execute_values and not cursor.executemany() or cursor.execute_batch()?

File: src/utils/connect_to_db.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import os
+import psycopg2
+from dotenv import load_dotenv
+#
+
+def connect_to_db():
```
Comment (dbrosatsmartworkz) [inline src/utils/connect_to_db.py]: Better to call this database_connection()

File: src/utils/insert_into_db.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        print("Insertion done correctly!")
+
+    except Exception as e:
+        print(f"Error inserting datas into the database: {e}")
+        return None
```
Comment (dbrosatsmartworkz) [inline src/utils/insert_into_db.py]: Unnecessary as the function returns None since no return values are stated for all options of the function

Comment (dbrosatsmartworkz) [reply src/utils/insert_into_db.py]: However I recommend returning True or False

File: src/utils/read_json_file.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import json
+
+def read_json_file(file_path):
+    with open(file_path, 'r') as file:
```
Comment (dbrosatsmartworkz) [inline src/utils/read_json_file.py:4]: What if the path doesnt exist

File: src/utils/insert_into_db.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        table_columns = [row[0] for row in cursor.fetchall()]
+
+        missing_columns = []
+        for col in table_columns:
+            if col not in column_order:
+                missing_columns.append(col)
+
+        if missing_columns:
+            print(f"Error: Missing mandatory column(s) in column_order: {missing_columns}")
+            return None
```
Comment (dbrosatsmartworkz) [inline src/utils/insert_into_db.py]: Return none on an error?

File: src/utils/connect_to_db.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        conn = psycopg2.connect(
+            dbname=os.getenv("DB_NAME"),
+            user=os.getenv("DB_USER"),
+            password=os.getenv("DB_PASSWORD"),
+        )
+        print("Successful connection!")
+        return conn
+    except Exception as e:
+        print(f"Error connecting to the database: {e}")
+        return None
```
Comment (dbrosatsmartworkz) [inline src/utils/connect_to_db.py]: Return none on an error?

**Review by dbrosatsmartworkz — COMMENTED** [review]
Some comments

File: db/meter_readings.sql (hunk 1 above)
Comment (JorgeAJT) [reply db/meter_readings.sql]: Yeah I just investigate about that and you're right, SERIAL is only on PostgreSQL and is older than int GENERATED ALWAYS AS IDENTITY. Thanks.

File: src/utils/insert_into_db.py (hunk 2 above)
Comment (JorgeAJT) [reply src/utils/insert_into_db.py]: I did to check every column was introduced in the input "column_order" of the function. If the person who called function forgot one column, the program will throw an error and it'll finish.

File: src/utils/insert_into_db.py (hunk 3 above)
Comment (JorgeAJT) [reply src/utils/insert_into_db.py]: It's the function I found to do that I wanted. Anyways I'll research about you said, thank you.

File: src/utils/read_json_file.py (hunk 6 above)
Comment (JorgeAJT) [reply src/utils/read_json_file.py:4]: Yeah, you're right, I have to create a, exception for that case

**Comment by dbrosatsmartworkz** [comment]
lgtm 👍

**Pull request merged**